### PR TITLE
Add Fancytech Fancyalice66

### PR DESCRIPTION
**Pull request by adophoxia**
## Description

A re-attempt of #17491, but updated to be more in line with current QMK standards, i.e, DD-configuration and removal of `action_exec` hack for encoders. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/fancytech/fancyalice66/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	    _______,          _______, _______,                   _______, _______,                   RGB_MOD,                   _______, RGB_VAD, _______
+    )
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
+    [0] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [1] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [2] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+    [3] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
```
Comment (zvecr) [inline keyboards/fancytech/fancyalice66/keymaps/via/keymap.c:57]: Has to match number of layers
```suggestion
    [1] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) }
```

File: keyboards/fancytech/fancyalice66/info.json [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	},
+	"diode_direction": "ROW2COL",
+	"matrix_pins": {
+		"rows": ["F4", "F5", "F6", "F7", "C7"],
+		"cols": ["C6", "E6", "B5", "B4", "D7", "D6", "D4", "D5", "D3", "D2", "D1", "D0", "B7", "B3", "B2", "B1"]
+	},
+	"layouts": {
+        "LAYOUT": {
+            "layout": [
+                { "matrix": [0, 0], "x": 0.75, "y": 0.5 },
```
Comment (fauxpark) [inline keyboards/fancytech/fancyalice66/info.json:82]:
```suggestion
                {"matrix": [0, 0], "x": 0.75, "y": 0.5},
```
etc...

File: keyboards/fancytech/fancyalice66/info.json [hunk 3]
```diff
@@ -0,0 +1,155 @@
+{
+	"keyboard_name": "FancyAlice66",
```
Comment (fauxpark) [inline keyboards/fancytech/fancyalice66/info.json:2]: Tabs to spaces please (4).

File: keyboards/fancytech/fancyalice66/info.json [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+	},
+	"encoder": {
+        "rotary": [
+            {"pin_a": "F0", "pin_b": "F1", "resolution": 2}
+        ]
+    },
+	"rgb_matrix": {
+		"driver": "WS2812",
+		"max_brightness": 200,
+		"animations":{
```
Comment (fauxpark) [inline keyboards/fancytech/fancyalice66/info.json:31]:
```suggestion
		"animations": {
```

File: keyboards/fancytech/fancyalice66/fancyalice66.c [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    //Key matrix (0 -> 65)
+	4,    1, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 1,
+	4, 4, 4, 4, 4, 4, 4, 4, 4,    4, 4, 4, 4, 4, 1,
+	4,    1, 4, 4, 4, 4, 4, 4,    4, 4, 4, 4, 4, 1,
+	   4,    4, 4, 4, 4, 4, 4,    4, 4, 4, 4, 4, 1,
+	4, 4, 4,       1,       4, 1,       4, 1,    1,
+
+    //Underglow (66 -> 81)
+    2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2
+}};
```
Comment (fauxpark) [inline keyboards/fancytech/fancyalice66/fancyalice66.c:50]: This can go to info.json.

File: keyboards/fancytech/fancyalice66/config.h [hunk 6]
```diff
@@ -0,0 +1,15 @@
+// Copyright 2022 chent7 (@chent7)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+/* RGB Matrix config */
+#define RGB_DI_PIN B0
```
Comment (fauxpark) [inline keyboards/fancytech/fancyalice66/config.h:7]: This can go to info.json (`rgblight.pin`, moving to `ws2812.pin` on `develop`).

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/fancytech/fancyalice66/info.json [hunk 7]
```diff
@@ -0,0 +1,155 @@
+{
+	"keyboard_name": "FancyAlice66",
+	"manufacturer": "Fancytech",
+	"url": "https://www.aliexpress.com/item/1005004272392611.html",
+	"maintainer": "Adophoxia",
+	"processor": "atmega32u4",
+	"bootloader": "caterina",
```
Comment (drashna) [inline keyboards/fancytech/fancyalice66/info.json:7]:
```suggestion
	"development_board": "promicro",
```

Comment (adophoxia) [reply keyboards/fancytech/fancyalice66/info.json:7]: Board doesn't use a physical Pro Micro and instead uses an embedded MCU.

File: keyboards/fancytech/fancyalice66/info.json (hunk 2 above)
Comment (adophoxia) [reply keyboards/fancytech/fancyalice66/info.json]: Resolved

File: keyboards/fancytech/fancyalice66/info.json (hunk 3 above)
Comment (adophoxia) [reply keyboards/fancytech/fancyalice66/info.json]: Resolved.

File: keyboards/fancytech/fancyalice66/info.json (hunk 4 above)
Comment (adophoxia) [reply keyboards/fancytech/fancyalice66/info.json]: Resolved.

File: keyboards/fancytech/fancyalice66/fancyalice66.c (hunk 5 above)
Comment (adophoxia) [reply keyboards/fancytech/fancyalice66/fancyalice66.c]: Moved into `info.json`

File: keyboards/fancytech/fancyalice66/config.h (hunk 6 above)
Comment (adophoxia) [reply keyboards/fancytech/fancyalice66/config.h]: Moved to `info.json`

File: keyboards/fancytech/fancyalice66/info.json [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+            {"flags": 2, "x": 149, "y": 0},
+            {"flags": 2, "x": 112, "y": 0},
+            {"flags": 2, "x": 75, "y": 0},
+            {"flags": 2, "x": 37, "y": 0},
+            {"flags": 2, "x": 0, "y": 0},
+            {"flags": 2, "x": 0, "y": 32},
+            {"flags": 2, "x": 0, "y": 64}
+        ],
+        "max_brightness": 200,
+        "rgblight.pin": "B0"
```
Comment (fauxpark) [inline keyboards/fancytech/fancyalice66/info.json:164]:
```suggestion
        "pin": "B0"
```

File: keyboards/fancytech/fancyalice66/info.json [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+    "maintainer": "Adophoxia",
+    "bootloader": "caterina",
+    "diode_direction": "ROW2COL",
+    "encoder": {
+        "rotary": [
+            {
+                "pin_a": "F0",
+                "pin_b": "F1",
+                "resolution": 2
+        	}
```
Comment (fauxpark) [inline keyboards/fancytech/fancyalice66/info.json:13]:
```suggestion
            {"pin_a": "F0", "pin_b": "F1", "resolution": 2}
```

File: keyboards/fancytech/fancyalice66/info.json [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+        ]
+	},
```
Comment (fauxpark) [inline keyboards/fancytech/fancyalice66/info.json:15]:
```suggestion
    },
```

File: keyboards/fancytech/fancyalice66/info.json [hunk 11]
```diff
@@ -0,0 +1,249 @@
...
+    "features": {
+        "bootmagic": true,
+        "encoder": false,
+        "extrakey": true,
+        "lto": true,
+        "mousekey": true,
+        "nkro": true,
+        "rgb_matrix": true
+	},
```
Comment (fauxpark) [inline keyboards/fancytech/fancyalice66/info.json:24]:
```suggestion
    },
```

File: keyboards/fancytech/fancyalice66/info.json [hunk 12]
```diff
@@ -0,0 +1,249 @@
...
+    "matrix_pins": {
+        "cols": ["C6", "E6", "B5", "B4", "D7", "D6", "D4", "D5", "D3", "D2", "D1", "D0", "B7", "B3", "B2", "B1"],
+        "rows": ["F4", "F5", "F6", "F7", "C7"]
+	},
```
Comment (fauxpark) [inline keyboards/fancytech/fancyalice66/info.json:28]:
```suggestion
    },
```

File: keyboards/fancytech/fancyalice66/info.json [hunk 13]
```diff
@@ -0,0 +1,249 @@
...
+            "solid_reactive_cross": true,
+            "solid_reactive_multicross": true,
+            "solid_reactive_multinexus": true,
+            "solid_reactive_multiwide": true,
+            "solid_reactive_nexus": true,
+            "solid_reactive_simple": true,
+            "solid_reactive_wide": true,
+            "solid_splash": true,
+            "splash": true
+    	},
```
Comment (fauxpark) [inline keyboards/fancytech/fancyalice66/info.json:72]:
```suggestion
        },
```

File: keyboards/fancytech/fancyalice66/info.json [hunk 14]
```diff
@@ -0,0 +1,249 @@
...
+            {"flags": 2, "x": 112, "y": 0},
+            {"flags": 2, "x": 75, "y": 0},
+            {"flags": 2, "x": 37, "y": 0},
+            {"flags": 2, "x": 0, "y": 0},
+            {"flags": 2, "x": 0, "y": 32},
+            {"flags": 2, "x": 0, "y": 64}
+        ],
+        "max_brightness": 200,
+        "rgblight.pin": "B0"
+	},
```
Comment (fauxpark) [inline keyboards/fancytech/fancyalice66/info.json:165]:
```suggestion
    },
```

File: keyboards/fancytech/fancyalice66/info.json [hunk 15]
```diff
@@ -0,0 +1,249 @@
...
+    "url": "https://www.aliexpress.com/item/1005004272392611.html",
+    "usb": {
+        "device_version": "0.0.1",
+        "force_nkro": true,
+        "pid": "0x3663",
+        "vid": "0xECED"
+	},
```
Comment (fauxpark) [inline keyboards/fancytech/fancyalice66/info.json:172]:
```suggestion
    },
```

File: keyboards/fancytech/fancyalice66/info.json [hunk 16]
```diff
@@ -0,0 +1,249 @@
...
+                {"matrix": [4, 6], "x": 7.25, "y": 4.5},
+                {"matrix": [4, 7], "w": 2.75, "x": 8.75, "y": 4.5},
+                {"matrix": [4, 10], "w": 1.5, "x": 11.5, "y": 4.5},
+                {"matrix": [4, 13], "x": 15.25, "y": 4.75},
+                {"matrix": [4, 14], "x": 16.25, "y": 4.75},
+                {"matrix": [4, 15], "x": 17.25, "y": 4.75}
+            ]
+    	}
+	}
+}
```
Comment (fauxpark) [inline keyboards/fancytech/fancyalice66/info.json:249]:
```suggestion
         }
    }
}
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/fancytech/fancyalice66/info.json [hunk 17]
```diff
@@ -0,0 +1,244 @@
...
+            {"flags": 2, "x": 187, "y": 0},
+            {"flags": 2, "x": 149, "y": 0},
+            {"flags": 2, "x": 112, "y": 0},
+            {"flags": 2, "x": 75, "y": 0},
+            {"flags": 2, "x": 37, "y": 0},
+            {"flags": 2, "x": 0, "y": 0},
+            {"flags": 2, "x": 0, "y": 32},
+            {"flags": 2, "x": 0, "y": 64}
+        ],
+        "max_brightness": 200,
```
Comment (fauxpark) [inline keyboards/fancytech/fancyalice66/info.json:159]:
```suggestion
        "max_brightness": 200
```

File: keyboards/fancytech/fancyalice66/fancyalice66.c [hunk 18]
```diff
@@ -0,0 +1,281 @@
...
+    if (clockwise) {
+        tap_code_delay(KC_VOLU, 10);
+    } else {
+        tap_code_delay(KC_VOLD, 10);
+    }
+    return true;
+}
+#endif
+
+{
```
Comment (fauxpark) [inline keyboards/fancytech/fancyalice66/fancyalice66.c:42]: This doesn't belong here.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/fancytech/fancyalice66/info.json (hunk 8 above)
Comment (adophoxia) [reply keyboards/fancytech/fancyalice66/info.json]: Resolved in [
16454d7](https://github.com/qmk/qmk_firmware/pull/20647/commits/16454d7897942d03de24058c14cf2cfe86efe766). Rest of commands follow suit.

File: keyboards/fancytech/fancyalice66/info.json [hunk 19]
```diff
@@ -0,0 +1,244 @@
...
+            {"flags": 2, "x": 149, "y": 0},
+            {"flags": 2, "x": 112, "y": 0},
+            {"flags": 2, "x": 75, "y": 0},
+            {"flags": 2, "x": 37, "y": 0},
+            {"flags": 2, "x": 0, "y": 0},
+            {"flags": 2, "x": 0, "y": 32},
+            {"flags": 2, "x": 0, "y": 64}
+        ],
+        "max_brightness": 200
+    },
```
Comment (fauxpark) [inline keyboards/fancytech/fancyalice66/info.json:160]:
```suggestion
    },
    "rgblight": {
        "pin": "B0"
    },
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/fancytech/fancyalice66/fancyalice66.c (hunk 18 above)
Comment (adophoxia) [reply keyboards/fancytech/fancyalice66/fancyalice66.c]: Removed the whole text of `info.json` from after the `#endif`

**Review by drashna — APPROVED** [review]

File: keyboards/fancytech/fancyalice66/fancyalice66.c [hunk 20]
```diff
@@ -0,0 +1,40 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "quantum.h"
+
+#ifdef RGB_MATRIX_ENABLE
+bool rgb_matrix_indicators_kb(void) {
```
Comment (waffle87) [inline keyboards/fancytech/fancyalice66/fancyalice66.c:20]:
```suggestion
bool rgb_matrix_indicators_kb(void) {
    if (!rgb_matrix_indicators_user() {
        return false;
    }
```

File: keyboards/fancytech/fancyalice66/fancyalice66.c [hunk 21]
```diff
@@ -0,0 +1,40 @@
...
+        return false; 
+    }
+    if (clockwise) {
+        tap_code_delay(KC_VOLU, 10);
+    } else {
+        tap_code_delay(KC_VOLD, 10);
+    }
+    return true;
+}
+#endif
```
Comment (waffle87) [inline keyboards/fancytech/fancyalice66/fancyalice66.c:40]: Will be the default behaviour on 2023-05-28
```suggestion
```

File: keyboards/fancytech/fancyalice66/info.json [hunk 22]
```diff
@@ -0,0 +1,247 @@
...
+    "rgblight": {
+        "pin": "B0"
+    },
```
Comment (waffle87) [inline keyboards/fancytech/fancyalice66/info.json:163]:
```suggestion
```

File: keyboards/fancytech/fancyalice66/config.h [hunk 23]
```diff
@@ -0,0 +1,12 @@
+// Copyright 2022 chent7 (@chent7)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#ifdef RGB_MATRIX_ENABLE
+    #define RGB_MATRIX_LED_COUNT 82
+
```
Comment (waffle87) [inline keyboards/fancytech/fancyalice66/config.h:8]:
```suggestion
#    define RGB_DI_PIN B0
#    define RGB_MATRIX_LED_COUNT 82
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/fancytech/fancyalice66/fancyalice66.c (hunk 20 above)
Comment (adophoxia) [reply keyboards/fancytech/fancyalice66/fancyalice66.c:20]: Tried to do this a couple of weeks ago and I ran into compile issue where it acted that `rgb_matrix_indicator_user` didn't exist. Will test again when I get back to my PC to make sure it works fine.

File: keyboards/fancytech/fancyalice66/config.h (hunk 23 above)
Comment (drashna) [reply keyboards/fancytech/fancyalice66/config.h:8]: At this point, to facilitate migration to ws2812 pin config.
```suggestion
#    define WS2812_DI_PIN B0
#    define RGB_DI_PIN WS2812_DI_PIN
#    define RGB_MATRIX_LED_COUNT 82

```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/fancytech/fancyalice66/config.h [hunk 24]
```diff
@@ -0,0 +1,14 @@
+// Copyright 2022 chent7 (@chent7)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#ifdef RGB_MATRIX_ENABLE
+#    define WS2812_DI_PIN B0
+#    define RGB_DI_PIN WS2812_DI_PIN
```
Comment (waffle87) [inline keyboards/fancytech/fancyalice66/config.h:8]:
```suggestion
#    define RGB_DI_PIN B0
```

**Review by waffle87 — APPROVED** [review]

**Pull request merged**